lib/fizzy/execute.cpp

```diff
@@ -573,10 +574,12 @@ ExecutionResult execute(
     const auto& code = instance.module->get_code(func_idx);
     auto* const memory = instance.memory.get();
 
-    const auto local_ctx = ctx.create_local_context();
+    const auto required_stack_space =
+        args_count + code.local_count + static_cast<size_t>(code.max_stack_height);
 
-    OperandStack stack(args, func_type.inputs.size(), code.local_count,
-        static_cast<size_t>(code.max_stack_height));
+    const auto local_ctx = ctx.create_local_context(required_stack_space);
```

Comment:
Should handle somehow that this can throw now.

Reply:
It throws before too.

Reply:
Ah right.

Reply:
To be precise: now `create_local_context()` throws, previously `OperandStack` was throwing (now `noexcept`).

Someone should handle this separately. But there is also easy solution: have fixed shared stack space. In case of out-of-space we can trap the execution.

Reply:
Catching `bad_alloc` and trapping would also be easy

Reply:
But this is non-deterministic execution.